maturity-level-one/src/Controllers/PlayersController.cs

```diff
@@ -74,10 +70,28 @@ public async Task<IActionResult> GetPlayer(int id)
         [SwaggerOperation("create-player")]
         [SwaggerResponse(201, "Created")]
         [SwaggerResponse(500, "API is not available")]
-        public async Task<IActionResult> Create(Player player)
+        public async Task<IActionResult> Create(NewPlayerDto player)
         {
-            await _worldCupRepository.CreatePlayer(player);
-            return CreatedAtRoute("get-player-byid", new { id = player.Id }, player);
+            var team = await _worldCupRepository.GetTeam(player.TeamId, includePlayers: false);
+
+            if (team == null)
+            {
+                return BadRequest($"The Team with Id {player.TeamId} does not exist.");
+            }
+
+            var playerEntity = new Player
```

Comment:
Discussion point: where's the line between using Automapper and a mapping like this?

Reply:
please open an issue to follow this discussion